File: 165-compare-version-numbers/wyrj/CompareVersionNumbers.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function(version1, version2) {
    var s1 = version1.split('.');
    var s2 = version2.split('.');
    var i, len = Math.max(s1.length, s2.length), n1, n2;
    for (i = 0; i < len; i++) {
        n1 = (i < s1.length) ? parseInt(s1[i], 10) : 0;
        n2 = (i < s2.length) ? parseInt(s2[i], 10) : 0;
        if (n1 > n2) return 1;
        if (n1 < n2) return -1;
    }
    return 0;
};

File: 165-compare-version-numbers/dennis/compare_version_numbers.py
class Solution(object):
    def append_zeros(self, input_list, expect_len):
        for i in xrange(expect_len - len(input_list)):
            input_list.append(0)

    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        v1_list = version1.split('.')
        v2_list = version2.split('.')
        max_len = max(len(v1_list), len(v2_list))
        self.append_zeros(v1_list, max_len)
        self.append_zeros(v2_list, max_len)

        for i in xrange(max_len):
            if int(v1_list[i]) > int(v2_list[i]):
                return 1
            elif int(v1_list[i]) < int(v2_list[i]):
                return -1

        return 0
